src/Feature/Editor/SurfaceView.re

```diff
@@ -73,24 +81,74 @@ let%component make =
 
   let {scrollX, scrollY, _}: Editor.t = editor;
 
-  let onMouseUp = (evt: NodeEvents.mouseButtonEventParams) => {
-    Log.trace("editorMouseUp");
-
+  let getMaybeLocationFromMousePosition = (mouseX, mouseY) => {
     maybeBbox^
-    |> Option.iter(bbox => {
+    |> Option.map(bbox => {
          let (minX, minY, _, _) = bbox |> BoundingBox2d.getBounds;
 
-         let relY = evt.mouseY -. minY;
-         let relX = evt.mouseX -. minX;
+         let relX = mouseX -. minX;
+         let relY = mouseY -. minY;
 
-         let (line, col) =
-           Editor.Slow.pixelPositionToBufferLineByte(
-             ~buffer,
-             ~pixelX=relX,
-             ~pixelY=relY,
-             editor,
-           );
+         Editor.Slow.pixelPositionToBufferLineByte(
+           ~buffer,
+           ~pixelX=relX,
+           ~pixelY=relY,
+           editor,
+         );
+       });
+  };
+
+  let onMouseMove = (evt: NodeEvents.mouseMoveEventParams) => {
+    getMaybeLocationFromMousePosition(evt.mouseX, evt.mouseY)
+    |> Option.iter(((line, col)) => {
+         dispatch(
+           Msg.MouseMoved({
+             location:
+               EditorCoreTypes.Location.create(
+                 Index.fromZeroBased(line),
+                 Index.fromZeroBased(col),
+               ),
+           }),
+         )
+       });
+    hoverTimerActive := true;
+    lastMousePosition := Some((evt.mouseX, evt.mouseY));
+    resetHoverTimer();
+  };
+
+  let%hook () =
+    Hooks.effect(
+      If(
+        (t, _) => Revery.Time.toFloatSeconds(t) > Constants.hoverTime,
+        hoverTimer,
+      ),
+      () => {
+        lastMousePosition^
+        |> Option.iter(((mouseX, mouseY)) => {
+             getMaybeLocationFromMousePosition(mouseX, mouseY)
+             |> Option.iter(((line, col)) => {
```

Comment:
This seems like it could be simplified:
```
lastMousePosition^
|> OptionEx.flatMap(((mouseX, mouseY)) => getMaybeLocationFromMousePosition(mouseX, mouseY))
|> Option.iter(((line, col)) => ...)
```
We can remove these nested iters by using `flatMap`/`bind`.